### PR TITLE
fix(ci): Re-add markdown and cue checks

### PR DESCRIPTION
**Pull request by jszwedko**
These were (accidentally?) removed in
https://github.com/vectordotdev/vector/pull/7368

Includes markdown fixes.

Closes: #9270

### CODE REVIEW

**Review by bruceg — APPROVED** [review]
Thanks @jszwedko

**Review by spencergilbert — APPROVED** [review]

**Comment by lucperkins** [comment]
As noted in my comments, overall I question the value of the Markdown linter in `website/content/*`. Removing this linter was deliberate, as I feel like it's going to produce unnecessary consternation (submit PR -> see error -> track down nit -> make change that doesn't seem quite right to satisfy linter).

File: website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md [hunk 1]
```diff
@@ -6,6 +6,6 @@ layout: component
 tags: ["aws", "cloudwatch", "component", "sink", "logs"]
 ---
 
-{{/* This doc is generated using:
+{{/*This doc is generated using:
```
[the remark this answers is not in the document]
Comment (lucperkins) [reply website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md:9]: In particular I'm referring to this:

```
{{/*This doc is generated using:
     1. The template in layouts/docs/component.html
2. The relevant CUE data in cue/reference/components/...*/}}
```

File: website/content/en/guides/advanced/cloudwatch-logs-firehose.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ tags: ["aws", "cloudwatch", "logs", "firehose", "advanced", "guides", "guide"]
 ---
 
 {{< requirement title="Pre-requisites" >}}
+
```
[the remark this answers is not in the document]
Comment (jszwedko) [reply website/content/en/guides/advanced/cloudwatch-logs-firehose.md:12]: I actually like the whitespace here, but honestly it doesn't matter to me as long as it is consistent which is the value that I think a linter provides.

File: website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jszwedko) [reply website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md:9]: Yeah, this is fair. If the linter doesn't gracefully Go templates, then I can see why we wouldn't want to use it for the Hugo markdown files. I'll see if I can fix this particular issue. That seems to me to be the only undesirable change in this PR.

[the remark this answers is not in the document]
Comment (jszwedko) [reply website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md]: I fixed this in a24132a which I personally think results in more readable comments.

[the remark this answers is not in the document]
Comment (lucperkins) [reply website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md]: @jszwedko Okay, I'm satisfied with that. If more edge cases like this pop up we could maybe disable the linter in a granular way (e.g. exclude `website/content/en/docs/reference/configuration/{sinks,sources,transforms}/*` but I think we're good for now.

**Review by lucperkins — APPROVED** [review]

**Comment by jszwedko** [comment]
> As noted in my comments, overall I question the value of the Markdown linter in `website/content/*`. Removing this linter was deliberate, as I feel like it's going to produce unnecessary consternation (submit PR -> see error -> track down nit -> make change that doesn't seem quite right to satisfy linter).

I think consternation is part of the role of a linter 😄 I personally see value in having consistently formatted cue and markdown, even in the website directory. The linter can be run locally when making changes.

We can keep an eye out to see if it starts causing a lot of pain with Go templates though.

File: website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jszwedko) [reply website/content/en/docs/reference/configuration/sinks/aws_cloudwatch_logs.md]: Thanks! Yeah, that makes sense, we can definitely keep an eye out for if it starts requiring less readable docs.

**Pull request merged**